libcontainer/container_linux.go

```diff
@@ -1545,6 +1556,59 @@ func (c *Container) Restore(process *Process, criuOpts *CriuOpts) error {
 	return err
 }
 
+// logCriuErrors tries to find and log errors from a criu log file.
+// The output is similar to what "grep -B5 Error" does.
+func logCriuErrors(dir, file string) {
+	lookFor := []byte("Error") // Print the line that contains this...
+	const before = 5           // ... and a few preceding lines.
+
+	logFile := filepath.Join(dir, file)
+	f, err := os.Open(logFile)
+	if err != nil {
+		logrus.Warn(err)
+		return
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	var lines [][]byte
+
+	lineNo := 0
+	lastLine := -1
+	didHeader := false
+	for s.Scan() {
+		lineNo++
```

Comment:
Would it make sense to use a limit for maximum number of lines?

Reply:
Right, I think we can stop after say 1000 lines or so.

Reply:
done

Reply:
@avagin says we're trying to solve a non-problem here. If criu was able to write this log file, and we waited for it to be written, we can definitely parse that one.

Removed the line limit